api/server/agent.go

```diff
@@ -22,9 +23,12 @@ func (tapi ToDDApi) Agent(w http.ResponseWriter, r *http.Request) {
 	// Retrieve query values
 	values := r.URL.Query()
 
-	var agent_list []defs.AgentAdvert
-
-	var tdb = db.NewToddDB(tapi.cfg)
+	agent_list, err := tapi.tdb.GetAgents()
```

Comment:
Ah, I forgot to add part of the explanation.

The signature of the function is `GetAgents() ([]defs.AgentAdvert, error)` implies a contract that it will either return a slice of AgentAdvert or an error. Since returning nil would violate that contract it must return an error in that case. So it's sufficient to simply check that error is nil. I believe this is a common convention, it would get pretty messy if we had to check both err and the value we want to retrieve for nil.